parcel_sourcemap_node/src/lib.rs

```diff
@@ -0,0 +1,581 @@
+
+    let mut napi_sources_array = ctx
+        .env
+        .create_array_with_length(source_map_instance.sources.len())?;
+    for (source_index, source) in source_map_instance.sources.iter().enumerate() {
+        napi_sources_array.set_element(source_index as u32, ctx.env.create_string(&source[..])?)?;
+    }
+
+    // Return array
+    return Ok(napi_sources_array);
+}
+
+#[js_function]
+fn get_sources(ctx: CallContext) -> Result<JsObject> {
+    return _get_sources(&ctx);
+}
+
+fn _get_sources_content(ctx: &CallContext) -> Result<JsObject> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &SourceMap = ctx.env.unwrap(&this)?;
+
+    let mut napi_sources_content_array = ctx
+        .env
+        .create_array_with_length(source_map_instance.sources_content.len())?;
+    for (source_index, source_content) in source_map_instance.sources_content.iter().enumerate() {
+        napi_sources_content_array.set_element(
+            source_index as u32,
+            ctx.env.create_string(&source_content[..])?,
+        )?;
+    }
+
+    // Return array
+    return Ok(napi_sources_content_array);
+}
+
+#[js_function]
+fn get_sources_content(ctx: CallContext) -> Result<JsObject> {
+    return _get_sources_content(&ctx);
+}
+
+#[js_function(1)]
+fn get_source_index(ctx: CallContext) -> Result<JsNumber> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index = source_map_instance.get_source_index(source.as_str()?)?;
+
+    match source_index {
+        Some(i) => {
+            return ctx.env.create_uint32(i);
+        }
+        None => {
+            return ctx.env.create_int32(-1);
+        }
+    }
+}
+
+#[js_function(2)]
+fn set_source_content_by_source(ctx: CallContext) -> Result<JsUndefined> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index: usize = source_map_instance.add_source(source.as_str()?) as usize;
+    let source_content = ctx.get::<JsString>(1)?.into_utf8()?;
+    source_map_instance.set_source_content(source_index, source_content.as_str()?)?;
+
+    return ctx.env.get_undefined();
+}
+
+#[js_function(1)]
+fn get_source_content_by_source(ctx: CallContext) -> Result<JsString> {
+    let this: JsObject = ctx.this_unchecked();
+    let source_map_instance: &mut SourceMap = ctx.env.unwrap(&this)?;
+
+    let source = ctx.get::<JsString>(0)?.into_utf8()?;
+    let source_index = source_map_instance.get_source_index(source.as_str()?)?;
+    match source_index {
+        Some(i) => {
+            let source_content = source_map_instance.get_source_content(i)?;
+            return ctx.env.create_string(source_content);
+        }
+        None => {
+            return ctx.env.create_string("");
```

Comment:
Should this throw an error or return null maybe?

Reply:
This is intentional to mimic the c++ logic.  Might make some breaking changes in a follow-up